src/AlertsManager.cpp

```diff
@@ -713,21 +739,59 @@ void AlertsManager::notifyAlert(AlertEntity alert_entity, const char *alert_enti
 
 /* **************************************************** */
 
+void AlertsManager::notifyToSlackIfNeeded (AlertEntity alert_entity, const char *alert_entity_value,
+        AlertType alert_type, AlertLevel alert_severity,
+        const char *alert_json,
+        const char *alert_origin, const char *alert_target) {
+
+  char buf[4], choice[32];
+  bool alert_to_be_notified=false; //indicates if the notification has to be done according to the slack notifications preference
+  SlackNotificationChoice notification_choice;
+
+  ntop->getTrace()->traceEvent(TRACE_NORMAL, 
+    "Entrata alert %s %s %s %s", getAlertLevel(alert_severity),
+    alert_entity_value, alert_origin, alert_target );
+
+  if((ntop->getRedis()->get((char*)ALERTS_MANAGER_NOTIFICATION_ENABLED,
```

Comment:
avoid Redis reads every time. Cache the values in a global prefs variable and use it. See this: https://github.com/SauroPollastrini/ntopng/blob/dev/include/Prefs.h#L154

make sure to refresh the variable when the user submit changes from the ntopng preferences UI.